grpc-proxy/data_converter.go

```diff
@@ -0,0 +1,67 @@
+package grpcproxy
+
+import (
+	"github.com/golang/snappy"
+	commonpb "go.temporal.io/api/common/v1"
+	"go.temporal.io/sdk/converter"
+)
+
+var DataConverter = NewDataConverter(converter.GetDefaultDataConverter())
+
+// NewDataConverter creates a new data converter that wraps the given data
+// converter with snappy compression.
+func NewDataConverter(underlying converter.DataConverter) converter.DataConverter {
+	return converter.NewCodecDataConverter(underlying, NewPayloadCodec())
+}
+
+func NewPayloadCodec() converter.PayloadCodec {
+	return &Codec{}
+}
+
+// Codec implements converter.PayloadEncoder for snappy compression.
+type Codec struct{}
+
+// Encode implements converter.PayloadCodec.Encode.
+func (e *Codec) Encode(payloads []*commonpb.Payload) ([]*commonpb.Payload, error) {
+	result := make([]*commonpb.Payload, len(payloads))
```

Comment:
_Technically_ to get the best compression, the input would be wrapped in a single `commonpb.Payloads`, that would be encrypted and returned back as a single-item slice (and the inverse on `Decode`). But I don't necessarily think it's a requirement.

Reply:
I don't think we should include such usage in samples yet as if for some reason it's copy/pasted into a system not using a proxy it could break Temporal Web for example, which never expects Payload count to change.